os/kernel/group/group_create.c

```diff
@@ -321,7 +324,13 @@ int group_initialize(FAR struct task_tcb_s *tcb)
 	irqrestore(flags);
 
 #endif
+#if defined(CONFIG_PREFERENCE) && CONFIG_TASK_NAME_SIZE > 0
+	/* The values of private preference are managed by task group.
+	 * Set a name of task group, tg_name for private preference. */
+
+	strncpy(group->tg_name, tcb->cmn.name, CONFIG_TASK_NAME_SIZE);
```

Comment:
Is it okay to use thread name as a preference identifier? What I worried is that thread name can be duplicated.